routes/weather.ts

```diff
@@ -259,13 +267,23 @@ export const getWateringData = async function( req: express.Request, res: expres
 			return;
 		}
 
-		wateringData = await weatherProvider.getWateringData( coordinates );
+		try {
+			wateringData = await weatherProvider.getWateringData( coordinates );
+		} catch ( err ) {
+			res.send( "Error: " + err );
+			return;
+		}
 	}
 
 	let scale = -1,	rainDelay = -1;
 
 	if ( adjustmentMethod === ADJUSTMENT_METHOD.ZIMMERMAN ) {
-		scale = calculateZimmermanWateringScale( adjustmentOptions, wateringData );
+		try {
+			scale = calculateZimmermanWateringScale( adjustmentOptions, wateringData );
+		} catch ( err ) {
+			res.send( "Error: " + err );
+			return;
+		}
 	}
 
 	if (wateringData) {
```

Comment:
We shouldnt need this check given try..catch above.

Reply:
`wateringData` can still be `undefined` if the manual adjustment method is being used

Reply:
You are right.

 I just don't like the logic in this area. It is too convoluted with nested ifs and `> METHOD.MANUAL` conditions that all feels a bit hard to decipher. Not one for this PR but I'll open an issue for consideration at a latter point.